qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -462,21 +462,35 @@ def simplify(self, atol: float | None = None, rtol: float | None = None) -> Spar
         """Simplify PauliList by combining duplicates and removing zeros.
 
         Args:
-            atol (float): Optional. Absolute tolerance for checking if
-                          coefficients are zero (Default: 1e-8).
-            rtol (float): Optional. relative tolerance for checking if
-                          coefficients are zero (Default: 1e-5).
+              atol (float): Optional. Absolute tolerance for checking if
+                            coefficients are zero (Default: 1e-8).
+              rtol (float): Optional. relative tolerance for checking if
+                            coefficients are zero (Default: 1e-5).
 
         Returns:
-            SparsePauliOp: the simplified SparsePauliOp operator.
+              SparsePauliOp: the simplified SparsePauliOp operator.
         """
         # Get default atol and rtol
         if atol is None:
             atol = self.atol
         if rtol is None:
             rtol = self.rtol
+        # ----- STEP 1: KEEP all terms (do NOT remove near-zero yet) -----
```

Comment:
The "step 1", "step 2" comments are perhaps a little unnecessary (and assigning numbers to things is a sure-fire way to let a comment get all out-of-sync as the file changes).  The "identify duplicate Paulis" and "sum coefficients of duplicate Paulis" comments are fine and useful - thanks for those.